scipy/sparse/_csr.py

```diff
@@ -122,6 +122,58 @@ def tobsr(self, blocksize=None, copy=True):
 
     tobsr.__doc__ = _spbase.tobsr.__doc__
 
+
+    def _broadcast_to(self, shape, copy=False):        
+        if self.shape == shape:
+            return self.copy() if copy else self
+        
+        shape = check_shape(shape, allow_nd=(self._allow_nd))
+
+        if np.broadcast_shapes(self.shape, shape) != shape:
+            raise ValueError("cannot be broadcast")
+        
+        if len(self.shape) == 1 and len(shape) == 1:
+            self.sum_duplicates()
+            if self.nnz == 0: # array has no non zero elements
+                return self.__class__(shape, dtype=self.dtype, copy=False)
+            
+            N = shape[0]
+            data = np.full(N, self.data[0])
+            indices = np.arange(0,N)
+            indptr = np.array([0, N])
+            return self.__class__((data, indices, indptr), shape=shape, copy=False)
+
+        # treat 1D as a 2D row
+        old_shape = self._shape_as_2d
+            
+        if len(shape) != 2:
+            raise ValueError('Target shape must be a tuple of length 2')
```

Comment:
Since `shape` can be 1D for CSR, the wording of the message can become softer:
```suggestion
            raise ValueError(f'CSR broadcast_to cannot have shape >2D. Got {ndim}D')```